lib/location/cloud_service/cloud_service_here_rest.c

```diff
@@ -220,16 +309,79 @@ static int location_service_generate_request(
 		       here_adjust_rsrq(cell_data->current_cell.rsrq),
 		       here_adjust_ta(cell_data->current_cell.timing_advance),
 		       neighbors);
+
 	if ((len < 0) || (len >= buf_len)) {
-		LOG_ERR("Too small buffer for HTTP request body");
-		return -ENOMEM;
+		LOG_ERR("Too small buffer for current cell with neighbors");
+		err = -ENOMEM;
 	}
+end:
+	k_free(neighbors);
+	return err;
+}
 
-	return 0;
+
+static int cloud_service_here_rest_pos_req_encode(
+	const struct lte_lc_cells_info *cell_data,
+	const struct wifi_scan_info *wifi_data,
+	char **buf)
+{
+	int err = 0;
+	char *cell_buf = NULL;
+	char *wifi_buf = NULL;
+
+	__ASSERT_NO_MSG(cell_data != NULL || wifi_data != NULL);
+	__ASSERT_NO_MSG(buf != NULL);
+
+	if (IS_ENABLED(CONFIG_LOCATION_METHOD_CELLULAR) && cell_data != NULL) {
+		err = cloud_service_here_rest_pos_req_encode_cellular(cell_data, &cell_buf);
+		if (err) {
+			LOG_ERR("Too small buffer for HTTP request body");
+			goto end;
+		}
+	}
+
+	if (IS_ENABLED(CONFIG_LOCATION_METHOD_WIFI) && wifi_data != NULL) {
+		err = cloud_service_here_rest_pos_req_encode_wifi(wifi_data, &wifi_buf);
+		if (err) {
+			LOG_ERR("Failed to generate Wi-Fi positioning request, err: %d", err);
+			goto end;
+		}
+	}
+
+	__ASSERT_NO_MSG(cell_buf != NULL || wifi_buf != NULL);
+	int cell_buf_len = (cell_buf != NULL) ? strlen(cell_buf) : 0;
+	int wifi_buf_len = (wifi_buf != NULL) ? strlen(wifi_buf) : 0;
+
+	/* 4 = '{' + ',' + '}' + '\0' */
```

Comment:
I don't see those extra characters added. Would `cell_buf_len + wifi_buf_len + 1` be sufficient?